examples/simple_trixi_controller.f90

```diff
@@ -25,18 +25,21 @@ program simple_trixi_controller_f
 
   ! Initialize Trixi
   call get_command_argument(1, argument)
-  call trixi_initialize(trim(argument) // c_null_char)
+  call trixi_initialize(argument)
 
   ! Set up the Trixi simulation
   ! We get a handle to use subsequently
   call get_command_argument(2, argument)
-  handle = trixi_initialize_simulation(trim(argument) // c_null_char)
+  handle = trixi_initialize_simulation(argument)
 
   ! Get time step length
   write(*, '(a, e14.8)') "Current time step length: ", trixi_calculate_dt(handle)
 
   ! Main loop
-  do while ( trixi_is_finished(handle) == 0 )
+  do
+    ! Exit loop once simulation is completed
+    if ( trixi_is_finished(handle) ) exit
+
```

Comment:
I have no knowledge here, but why would you not use the `do while (<condition>)` syntax?

Reply:
From [Modern Fortran Explained: Incorporating Fortran 2018](https://doi.org/10.1093/oso/9780198811893.001.0001):
> An alternative, but redundant, form of [the `do` construct] is its representation using a `do while` statement [...]. We prefer the form that uses the `exit` statement because this can be placed anywhere in the loop, whereas the `do while` statement always performs its test at the loop start.

But we can also leave this as-is if you prefer.

Reply:
I have no preference. My question was just out of curiosity.